engine/src/star_system.cpp

```diff
@@ -918,8 +918,7 @@ void StarSystem::UpdateUnitsPhysics(bool firstframe) {
         try {
             UnitCollection col = physics_buffer[current_sim_location];
             un_iter iter = physics_buffer[current_sim_location].createIterator();
-            Unit *unit = nullptr;
-            for (; (unit = *iter); ++iter) {
+            for (Unit *unit = nullptr; (unit = *iter); ++iter) {
```

Comment:
Add a TODO here and create an issue please - this is a prime candidate for multi-threading. NM. Opened #992.

Reply:
I emphatically agree. Though we will need to be careful in how we implement it. We'll probably have to have some sort of locking mechanism so that two threads don't do conflicting things with the same Unit at the same time.

But yes, this is definitely something I want to look at figuring out in the near future. The physics calculations are by far the most time-consuming part of the game. Making that multi-threaded would be a boon for performance.